### PR TITLE
fix(solc): add `RuntimeOrHandle` & fix solc blocking installation

### PR DESCRIPTION
**Pull request by rkrasiuk**
## Motivation

#1259 

## PR Checklist

- [x] Added Tests
- [x] Added Documentation
- [ ] Updated the changelog

### CODE REVIEW

File: ethers-solc/Cargo.toml [hunk 1]
```diff
@@ -19,7 +19,7 @@ serde_json = "1.0.68"
 serde = { version = "1.0.130", features = ["derive"] }
 semver = { version = "1.0.9", features = ["serde"] }
 walkdir = "2.3.2"
-tokio = { version = "1.15.0", default-features = false, features = ["process", "io-util", "fs", "time"], optional = true }
+tokio = { version = "1.15.0", default-features = false, features = ["process", "io-util", "fs", "time"] }
```
Comment (mattsse) [inline ethers-solc/Cargo.toml:22]: I don't think we need all features here just for the executor?

File: ethers-solc/src/compile/mod.rs [hunk 2]
```diff
@@ -371,9 +371,11 @@ impl Solc {
     /// Blocking version of `Self::install`
     #[cfg(all(feature = "svm-solc"))]
     pub fn blocking_install(version: &Version) -> std::result::Result<Self, svm::SolcVmError> {
```
Comment (mattsse) [inline ethers-solc/src/compile/mod.rs:373]: let's add some docs why we're doing this

File: ethers-solc/src/compile/mod.rs [hunk 3]
```diff
@@ -799,6 +801,17 @@ mod tests {
...
     }
 
+    #[test]
+    #[cfg(feature = "svm-solc")]
+    fn can_install_solc_in_tokio_rt() {
+        let ver = "0.8.6";
+        let version = Version::from_str(ver).unwrap();
+        let result = tokio::runtime::Runtime::new()
+            .unwrap()
+            .block_on(async { Solc::blocking_install(&version) });
```
Comment (mattsse) [inline ethers-solc/src/compile/mod.rs:811]: great!

can we also add some integration tests that pin a specific solc version and make the tests `#[tokio::test] async`?

**Review by mattsse — CHANGES_REQUESTED** [review]
great, let's add some additional docs and project tests

**Review by mattsse — CHANGES_REQUESTED** [review]
this currently fails on wasm due to tokio, but `reqwest::blocking` 

lets do some cfgs to use `svm::blocking_install` directly in wasm

**Review by mattsse — APPROVED** [review]
lgtm

**Review by gakonst — APPROVED** [review]

**Pull request merged**